### PR TITLE
Add deleteAncientTrend unit testing for trends.js

### PR DESCRIPTION
**Pull request by ntarn**
Add deleteAncientTrend unit testing for trends.js

### CODE REVIEW

File: search-sentiments-map/test/trends-test.js [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+const datastore = new Datastore();
+
+describe('Trends', function() {
+  describe('DeleteAncientTrend', function() {
+    let datastoreEntities;
+    const CURRENT_TIME = Date.now();
+    const STALE_DATA_THRESHOLD_7_DAYS_MS = 7 * 24 * 60 * 60000;
+    const THRESHOLD_6_DAYS_MS = 6 * 24 * 60 * 60000;
+    const DIFFERENCE_CURRENT_TIME_THRESHOLD_6_DAYS_MS =
+        CURRENT_TIME - THRESHOLD_6_DAYS_MS;
```
Comment (hayleejane3) [inline search-sentiments-map/test/trends-test.js:30]: alphabetical order

File: search-sentiments-map/test/trends-test.js [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+        // June 30 2020. Older than the Threshold.
+        {timestamp: 1593541470000},
+        {timestamp: Date.now()}
+      ];
+      datastoreEntities[0][datastore.KEY] =
+          datastore.key(['TrendsEntry', 0]);
+      datastoreEntities[1][datastore.KEY] =
+          datastore.key(['TrendsEntry', 1]);
+      datastoreEntities[2][datastore.KEY] =
+          datastore.key(['TrendsEntry', 2]);
```
Comment (hayleejane3) [inline search-sentiments-map/test/trends-test.js:44]: foreach

File: search-sentiments-map/test/trends-test.js [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+        {timestamp: Date.now()},
+        {timestamp: Date.now()}
+      ];
+
+      datastoreEntities[0][datastore.KEY] =
+          datastore.key(['TrendsEntry', 0]);
+      datastoreEntities[1][datastore.KEY] =
+          datastore.key(['TrendsEntry', 1]);
+      datastoreEntities[2][datastore.KEY] =
+          datastore.key(['TrendsEntry', 2]);
```
Comment (hayleejane3) [inline search-sentiments-map/test/trends-test.js:57]: Here as well

File: search-sentiments-map/test/trends-test.js [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+
+      // Stub calls to the datastore.
+      sinon.stub(Datastore.prototype, 'runQuery').callsFake(() => {
+        return [datastoreEntities];
+      });
+
+      // Stub calls to the datastore.
+      sinon.stub(Datastore.prototype, 'delete').callsFake((entity) => {
```
Comment (hayleejane3) [inline search-sentiments-map/test/trends-test.js:65]:
```suggestion
      });
      sinon.stub(Datastore.prototype, 'delete').callsFake((entity) => {
```

File: search-sentiments-map/test/trends-test.js [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+      });
+    });
+
+    afterEach(() => {
+      sinon.restore();
+    });
+
+    it('should delete the oldest trend datastore entity', async function() {
+      await trends.deleteAncientTrend();
+      // Should delete the June 29 2020 entity.
```
Comment (hayleejane3) [inline search-sentiments-map/test/trends-test.js:76]:
```suggestion
      // Verify that the June 29 2020 entity is deleted.
```

File: search-sentiments-map/test/trends-test.js [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+      sinon.restore();
+    });
+
+    it('should delete the oldest trend datastore entity', async function() {
+      await trends.deleteAncientTrend();
+      // Should delete the June 29 2020 entity. 
+      assert.equal(datastoreEntities.length, 2);
+      // The first entity should be the June 30 2020 entity.
+      assert.equal(datastoreEntities[0].timestamp, 1593541470000);
+      // June 30, 2020 is still older than the threshold.
```
Comment (hayleejane3) [inline search-sentiments-map/test/trends-test.js:80]: Update these comments to follow the general pattern for test comments like above.

File: search-sentiments-map/test/trends-test.js [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+      assert.isAbove(Date.now() - datastoreEntities[0].timestamp,
+          STALE_DATA_THRESHOLD_7_DAYS_MS);
+    });
+  });
+  describe('DeleteAncientTrendWithNoTrendDeleted', function() {
+    let datastoreEntitiesBelowThreshold;
+    const CURRENT_TIME = Date.now();
+    const STALE_DATA_THRESHOLD_7_DAYS_MS = 7 * 24 * 60 * 60000;
+    const THRESHOLD_6_DAYS_MS = 6 * 24 * 60 * 60000;
+    const DIFFERENCE_CURRENT_TIME_THRESHOLD_6_DAYS_MS =
```
Comment (hayleejane3) [inline search-sentiments-map/test/trends-test.js:90]: alphabetical order

File: search-sentiments-map/test/trends-test.js [hunk 8]
```diff
@@ -0,0 +1,133 @@
...
+          datastore.key(['TrendsEntry', 1]);
+      datastoreEntitiesBelowThreshold[2][datastore.KEY] =
+          datastore.key(['TrendsEntry', 2]);
+
+      // Stub calls to the datastore.
+      sinon.stub(Datastore.prototype, 'runQuery').callsFake(() => {
+        return [datastoreEntitiesBelowThreshold];
+      });
+
+      // Stub calls to the datastore.
```
Comment (hayleejane3) [inline search-sentiments-map/test/trends-test.js:111]:
```suggestion
        return [datastoreEntitiesBelowThreshold];
      });
```

File: search-sentiments-map/test/trends-test.js [hunk 9]
```diff
@@ -0,0 +1,133 @@
...
+
+    it('should delete nothing because timestamps are below the threshold', 
+        async function() {
+      await trends.deleteAncientTrend();
+      assert.equal(datastoreEntitiesBelowThreshold.length, 3);
+      // The first entity should be the oldest entity.
+      assert.equal(datastoreEntitiesBelowThreshold[0].timestamp, 
+          DIFFERENCE_CURRENT_TIME_THRESHOLD_6_DAYS_MS);
+      // The first entity is not older than the threshold.
+      assert.isAtMost(CURRENT_TIME - datastoreEntitiesBelowThreshold[0].timestamp,
```
Comment (hayleejane3) [inline search-sentiments-map/test/trends-test.js:129]: Comments here too

File: search-sentiments-map/test/trends-test.js (hunk 2 above)
Comment (ntarn) [reply search-sentiments-map/test/trends-test.js:44]: i did a for loop for both bc i need to access the index

File: search-sentiments-map/routes/trends.js [hunk 10]
```diff
@@ -256,5 +256,10 @@ function getGlobalTrends(trendsByCountry) {
...
 
+// Necessary for unit testing.
+const trends = {
+  deleteAncientTrend,
+}
+
 module.exports.router = router;
-module.exports.updateTrendsFunction = updateDailyTrends;
\ No newline at end of file
+module.exports.trends = trends;
```
Comment (carmenBen) [inline search-sentiments-map/routes/trends.js:265]: If you are changing this here, be sure to also change the update-data.js file to call trends.updateTrendsFunction, and also add updateTrendsFunction into your trends const right above. Alternatively, don't delete the updateTrends export, and handle when you merge in.

Comment (ntarn) [reply search-sentiments-map/routes/trends.js:265]: yea I deleted it bc i saw in chenyu's PR that she had updated these kind of things so I was going to merge after her PR is approved

Comment (carmenBen) [reply search-sentiments-map/routes/trends.js:265]: Okay soudns good!

**Pull request merged**